compiler/src/dmd/declaration.d

```diff
@@ -1520,12 +1520,18 @@ extern (C++) class VarDeclaration : Declaration
             }
             else
             {
-                // __ArrayDtor(v[0 .. n])
+                // __ArrayDtor((cast(T[n])v)[0 .. n])
                 e = new VarExp(loc, this);
 
                 const sdsz = sd.type.size();
                 assert(sdsz != SIZE_INVALID && sdsz != 0);
                 const n = sz / sdsz;
+
+                Type sty = sd.type.sarrayOf(n);
```

Comment:
Why do we need this step?

Reply:
Because leaving it be will trigger a errormessage for the SliceExp since `n` is the full length of the multidimensional array (e.g. 4 when using `T[2][2]`), which can be very confusing for users since they typically dont (need to) know that D injects calls to `__ArrayDtor` at the end of the function. Casting it beforehand to a static array of size `n` fixes this.

Reply:
Can this be a comment.

Reply:
> Can this be a comment.

What do you mean by that?

Reply:
Future travellers reading the source code may ask the same question.

Reply:
> Future travellers reading the source code may ask the same question.

Ohhh you mean to add the explanation as a comment to the code! Sure can do :)